client/src/components/MapElement/MapElement.js

```diff
@@ -25,15 +40,61 @@ export function MapElement({ className }) {
     )
   }, [appView, activeTripId])
 
-  const bgUrl = useMemo(() => {
-    if (appView === AppView.NEW_TRIP) return "bg-[url('../public/globe.png')]"
-    return "bg-[url('https://assets.website-files.com/5e832e12eb7ca02ee9064d42/5f7db426b676b95755fb2844_Group%20805.jpg')]"
-  }, [appView])
+  const addMarkerOnMap = (longitude, latitude) => {
+    const markerElement = document.createElement('div')
+    markerElement.innerHTML =
+      '<img src="https://cdn-icons-png.flaticon.com/512/1670/1670080.png" alt="Marker Icon" style="width: 40px; height: 40px;"/>'
+    const marker = new mapboxgl.Marker(markerElement)
+      .setLngLat([longitude, latitude])
+      .addTo(map)
+    const updatedMarkersOnMap = [...markersOnMap, marker]
+    setMarkersOnMap(updatedMarkersOnMap)
+  }
+
+  const clearMarkersOnMap = () => {
+    markersOnMap.forEach((marker) => marker.remove())
+    setMarkersOnMap([])
+  }
+
+  const flyToLocation = useCallback(
+    (longitude, latitude, zoom) => {
+      map.flyTo({
+        center: [longitude, latitude],
+        zoom,
+        essential: true,
+      })
+    },
+    [map],
+  )
+
+  useEffect(() => {
+    clearMarkersOnMap()
+    for (let markerWithProps of markersWithProps) {
+      addMarkerOnMap(markerWithProps.longitude, markerWithProps.latitude)
+    }
+  }, [markersWithProps])
```

Comment:
see Github actions. btw before you commit, does vscode tell you issues it has found? please check those before committing